pkg/remoteconfig/orchestrator_k8s_crd.go

```diff
@@ -0,0 +1,136 @@
+package remoteconfig
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-operator/api/datadoghq/v2alpha1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// CustomResourceDefinitionURLs defines model for CustomResourceDefinitionURLs.
+type CustomResourceDefinitionURLs struct {
+	Crds *[]string `json:"crds,omitempty"`
+}
+
+func (r *RemoteConfigUpdater) crdConfigUpdateCallback(updates map[string]state.RawConfig, applyStatus func(string, state.ApplyStatus)) {
+	ctx := context.Background()
+
+	var configIDs []string
+	for id := range updates {
+		applyStatus(id, state.ApplyStatus{State: state.ApplyStateUnacknowledged, Error: ""})
+		configIDs = append(configIDs, id)
+	}
+
+	mergedUpdate, err := r.parseCRDReceivedUpdates(updates, applyStatus)
+	if err != nil {
+		r.logger.Error(err, "Failed to merge updates")
+		return
+	}
+
+	if err := r.getAndUpdateDatadogAgentWithRetry(ctx, mergedUpdate, r.crdUpdateInstanceStatus); err != nil {
+		r.logger.Error(err, "Failed to update status")
+		applyStatus(configIDs[len(configIDs)-1], state.ApplyStatus{State: state.ApplyStateError, Error: err.Error()})
+		return
+	}
+
+	// Acknowledge that configs were received
+	for _, id := range configIDs {
+		applyStatus(id, state.ApplyStatus{State: state.ApplyStateAcknowledged, Error: ""})
+	}
+
+	r.logger.Info("Successfully applied configuration")
+
+}
+
+func (r *RemoteConfigUpdater) parseCRDReceivedUpdates(updates map[string]state.RawConfig, applyStatus func(string, state.ApplyStatus)) (DatadogAgentRemoteConfig, error) {
+	// Unmarshal configs and config order
+	crds := []string{}
+	for _, c := range updates {
+		if c.Metadata.Product == state.ProductOrchestratorK8sCRDs {
+			rcCRDs := CustomResourceDefinitionURLs{}
+			err := json.Unmarshal(c.Config, &rcCRDs)
+			if err != nil {
+				return DatadogAgentRemoteConfig{}, err
+			}
+			if rcCRDs.Crds != nil {
+				crds = append(crds, *rcCRDs.Crds...)
+			}
+		}
+	}
+
+	if len(crds) == 0 {
+		r.logger.Info("No CRDs received")
+		return DatadogAgentRemoteConfig{}, nil
+	}
+
+	// Merge configs
+	var finalConfig DatadogAgentRemoteConfig
+
+	// Cleanup CRD duplicates and add to final config
+	crds = removeDuplicateStr(crds)
+	if finalConfig.ClusterAgent == nil {
+		finalConfig.ClusterAgent = &ClusterAgentFeaturesConfig{}
+	}
+	if finalConfig.ClusterAgent.CRDs == nil {
+		finalConfig.ClusterAgent.CRDs = &CustomResourceDefinitionURLs{}
+	}
+	finalConfig.ClusterAgent.CRDs.Crds = &crds
```

Comment:
Current implementation for security products unmarshals `configuration_order`  and uses to merge multiple configs:
https://github.com/DataDog/datadog-operator/blob/03611c4e0472e9f993aa07d30b9fe051207a2fc9/pkg/remoteconfig/updater.go#L323-L324

do we need similar logic for CRDs too?